# Principles of Turbomachinery, S. A. Korpela, Wiley, 2019 (2nd Edition): Problem 3.9

Air flows subsonically in a converging nozzle. At a downstream location where the Mach number is $M_2 = 0.5$, the flow area is $A_2$. What is the Mach number at an upstream location where $A_1 = 2 A_2$.

---

In [12]:
from sympy import Symbol
from sympy.solvers import solve

# Problem data
M_2 = 0.5
gamma = 1.4

Let us relate location with flow are $A_2$ to the sonic location:

$$
\frac{A_2}{A^*} = \frac{1}{M_2} \left( \frac{2}{\gamma + 1} + \frac{\gamma - 1}{\gamma + 1} M_2^2 \right)^{\frac{\gamma + 1}{2 (\gamma - 1)}}
$$

We can then find $A^*$ as follows:

$$
A^* = \frac{A_2}{r}
$$

In [10]:
r = 1/M_2 * (2/(gamma + 1) + (gamma - 1)/(gamma + 1) * M_2 ** 2) ** ((gamma + 1)/(2 * (gamma - 1)))
print(f"r = {r:0.2f}")

r = 1.34


Similarly, let us relate $A^*$ to $A_1$ (using the realtion $A_1 = 2 A_2$):

$$
2 \frac{A_2}{A^*} = \frac{1}{M_1} \left( \frac{2}{\gamma + 1} + \frac{\gamma - 1}{\gamma + 1} M_1^2 \right)^{\frac{\gamma + 1}{2 (\gamma - 1)}}
$$

In [14]:
M_1 = Symbol('M_1', real = True, positive = True)
sol = solve(2 * r - 1/M_1 * (2/(gamma + 1) + (gamma - 1)/(gamma + 1) * M_1 ** 2) ** ((gamma + 1)/(2 * (gamma - 1))), M_1)
sol

[0.222434050020798, 2.51730370340075]

The solution to the quadratic equations yields two solutions, both with physical significance. We know, however, that the nozzle is subsonic. Therefore the correct solution is 0.222